### PR TITLE
Add codecov to the github actions

### PR DESCRIPTION
**Pull request by blayson**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by epapbak — APPROVED** [review]
lgtm

File: .github/workflows/run_tests.yaml [hunk 1]
```diff
@@ -19,7 +19,15 @@ jobs:
         run: docker-compose build vuln4shift_unit_tests
 
       - name: Run unit tests
-        run: docker-compose run --rm vuln4shift_unit_tests
```
Comment (jdobes) [inline .github/workflows/run_tests.yaml:22]: If there is a local dir mount, the `--rm` can stay here.

File: docker-compose.yml [hunk 2]
```diff
@@ -82,6 +82,8 @@ services:
         condition: service_completed_successfully
     profiles:
       - unit-tests
+    volumes:
+      - ./:/vuln4shift
```
Comment (jdobes) [inline docker-compose.yml:86]: @yungbender will there be needed any changes in building the test image if we just mount the git dir (like removing some unnecessary steps)? Or is it ok?

File: .github/workflows/run_tests.yaml (hunk 1 above)
Comment (jdobes) [reply .github/workflows/run_tests.yaml:22]: But it doesn't really matter, the container will be removed once the GH action ends anyway.

**Review by jdobes — APPROVED** [review]

**Pull request merged**